hydromt/workflows/basin_mask.py

```diff
@@ -15,18 +15,18 @@
 from shapely.geometry import box
 
 from ..data_adapter import GeoDataFrameAdapter
+from ..data_catalog import DataCatalog
 from ..flw import basin_map, flwdir_from_da, outlet_map, stream_map
 
 # local
-from ..io import open_raster
 from ..models import MODELS
 
 logger = logging.getLogger(__name__)
 
 __all__ = ["get_basin_geometry", "parse_region"]
 
 
-def parse_region(region, logger=logger):
+def parse_region(region, logger=logger, data_catalog=DataCatalog()):
```

Comment:
putting a new datacatalog as a default argument is going to create a new one every time this is parsed, but not every time the funciton gets called, which makes the control flow very confusing. If you want to keep a default argument like this, you should set it to `None` and make a new object in the case this is None. Personally I'd be in favor of making the data_catalog a required argument.